### PR TITLE
Updates ViewConfig types to delegate isInAParentText context

### PR DESCRIPTION
**Pull request by rozele**
## Summary

Now that HostContext determination for Fabric is a DEV-only behavior, we can move the HostContext determination to resolve from the ViewConfig for a given type. Doing this will allow arbitrary types to register themselves as potential parents of raw text string children. This is the first of two diffs for react as we'll:

1. Add the new property to the ViewConfig types
2. Update React Native to include the `supportsRawText` property for `RCTText`, `RCTVirtualText`, `AndroidTextInput`, etc.
3. Switch the behavior of react to read from the ViewConfig rather than a static list of types.

## How did you test this change?

- yarn test
- yarn test --prod
- Pulled change into react-native, added `supportsRawText` props to RCTText, RCTVirtualText, etc. ViewConfigs and confirmed everything type checks.

### CODE REVIEW

File: packages/react-native-renderer/src/ReactFiberConfigFabric.js [hunk 1]
```diff
@@ -228,12 +231,14 @@ export function getChildHostContext(
...
 ): HostContext {
   const prevIsInAParentText = parentHostContext.isInAParentText;
-  const isInAParentText =
-    type === 'AndroidTextInput' || // Android
-    type === 'RCTMultilineTextInputView' || // iOS
-    type === 'RCTSinglelineTextInputView' || // iOS
-    type === 'RCTText' ||
-    type === 'RCTVirtualText';
+  const isInAParentText = enableViewConfigParentTextContext
+    ? getViewConfigForType(type).supportsRawText
```
Comment (motiz88) [inline packages/react-native-renderer/src/ReactFiberConfigFabric.js:235]: Probably want to coalesce this to `false` if it's not normalised elsewhere.

```suggestion
    ? getViewConfigForType(type).supportsRawText ?? false
```

File: packages/react-native-renderer/src/ReactNativeTypes.js [hunk 2]
```diff
@@ -92,6 +92,7 @@ export type ViewConfig = $ReadOnly<{
 export type PartialViewConfig = $ReadOnly<{
   bubblingEventTypes?: $PropertyType<ViewConfig, 'bubblingEventTypes'>,
   directEventTypes?: $PropertyType<ViewConfig, 'directEventTypes'>,
+  supportsRawText?: boolean,
```
Comment (motiz88) [inline packages/react-native-renderer/src/ReactNativeTypes.js:95]: Looks like this should also be added to the `ViewConfig` type - possibly as optional for now, then required when we've updated React Native to always supply it.

**Review by javache — DISMISSED** [review]
Could we make this change incremental? Eg check for `supportsRawText`, and if not set or false, check the existing list of component names? That way we have to do less juggling to make this change atomic.

**Review by rickhanlonii — CHANGES_REQUESTED** [review]
Can you add tests? I know that this just updates the types, but there should be some sort of tests for this behavior before we start implementing it. And the existing tests that are there are currently passing, but should fail without the changes @javache mentioned.

Here's where I would add them:  https://github.com/facebook/react/blob/1940cb27b260c2eab79c76763d1151ba18353ff8/packages/react-native-renderer/src/__tests__/ReactFabric-test.internal.js#L583

**Comment by rozele** [comment]
> Could we make this change incremental? Eg check for `supportsRawText`, and if not set or false, check the existing list of component names? That way we have to do less juggling to make this change atomic.

Not seeing the value in this, it's just as easy to:
1. expose the types in React
2. make the change in RN to use the types
3. Switch to using ViewConfig in React

As it is to:
1. Expose the types and using ViewConfig with fallback in React
2. Make the change in RN to use the types
3. Remove the fallback list in React

**Comment by rozele** [comment]
@rickhanlonii if we are just adding types in this change, there's not really anything to test. The subsequent change after RN is updated to consume the new types will need to update the existing tests, which appear to have sufficient coverage.

**Comment by rickhanlonii** [comment]
> here's not really anything to test

Yeah ok sure, we can add tests when we do step 3 to actually consume the prop in React. If you're going this route though, it would be easier to just add the props to the config with a FlowFixMe and remove the FlowFixMe after the behavior change lands. 

Then you could add the type, the behavior, and the tests all in one React PR.

**Review by rickhanlonii — APPROVED** [review]

**Pull request merged**